src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -99,10 +124,17 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     ) {
       throw new Error("Parameter geo_location_sources needs to be an array");
     }
-
-    this._config = config;
-    this._configEntities = config.entities
-      ? processConfigEntities<MapEntityConfig>(config.entities)
+    if (config.show_all && (config.entities || config.geo_location_sources)) {
+      throw new Error(
+        "Cannot specify show_all and entities or geo_location_sources"
+      );
+    }
+    this._config = { ...config };
+    if (this.hass && config.show_all) {
+      this._config.entities = this._getAllEntities();
```

Comment:
Feels a bit odd but I guess there's nothing wrong with a card modifying its own config?

I think this is needed so `hasConfigOrEntitiesChanged` tracks these auto-entities correctly.

Reply:
Should be fine. You're even cloning the config object so I can't see how anything can go wrong